### PR TITLE
fix err highlighting by converting into valid json, remove proprietary code, change markdown engine and upgrade to the latest version of jekyll

### PR DESCRIPTION
**Pull request by mzruya**
image

made the following changes
1. ```javascript                       => ```json
2. -{% highlight javascript %} => ```json
3. " :                                         => ":
4. unindented all the json examples to start at column 0
5. fixed json examples to have commas when needed

@robbiegill, it might help if we compare it against `docs.gusto.com` together for pixel precision :-)

### CODE REVIEW

File: _plugins/redcarpet2_markdown.rb [hunk 1]
```diff
@@ -1,48 +0,0 @@
-require 'fileutils'
-require 'digest/md5'
-require 'redcarpet'
-require 'albino'
```
Comment (mzruya) [inline _plugins/redcarpet2_markdown.rb]: Alibno is no longer maintained (https://github.com/github-archive/albino), this is the only thing that uses it, i'm not sure why it's needed. removing this allows me to remove Alibno from our gem file and upgrade to the latest everything

File: Gemfile [hunk 2]
```diff
@@ -1,6 +1,5 @@
 source 'https://rubygems.org'
 
 # locked to github pages versions
-gem 'jekyll',     '=1.1.2'
-gem 'rdiscount',  '=1.6.8'
-gem 'albino'
\ No newline at end of file
+gem 'jekyll',     '=3.2.1'
+gem 'kramdown',  '=1.11.1'
```
Comment (mzruya) [inline Gemfile]: https://pages.github.com/versions/
https://github.com/blog/2100-github-pages-now-faster-and-simpler-with-jekyll-3-0

```
Starting May 1st, 2016, GitHub Pages will only support kramdown, Jekyll's default Markdown engine. If you are currently using Rdiscount or Redcarpet we've enabled kramdown's GitHub-flavored Markdown support by default, meaning kramdown should have all the features of the two deprecated Markdown engines, so the transition should be as simple as updating the Markdown setting to kramdown in your site's configuration (or removing it entirely) over the course of the next three months.
```

File: _config.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
-markdown: redcarpet
-pygments: true
+markdown: kramdown
+highlighter: rouge
 permalink: /docs/:version/:resource
 redcarpet:
   extensions: ["no_intra_emphasis", "fenced_code_blocks", "autolink", "strikethrough", "superscript", "with_toc_data", "tables"]
```
Comment (mzruya) [inline _config.yml]: i wonder if we still need this?

File: _config.yml [hunk 4]
```diff
@@ -1,5 +1,3 @@
-markdown: redcarpet
-pygments: true
+markdown: kramdown
+highlighter: rouge
 permalink: /docs/:version/:resource
-redcarpet:
-  extensions: ["no_intra_emphasis", "fenced_code_blocks", "autolink", "strikethrough", "superscript", "with_toc_data", "tables"]
```
Comment (mzruya) [inline _config.yml]: everything seems to look and work the same, we can smoke test together :-)

**Comment by robbiegill** [comment]
👍 was great for me! - I pushed a couple small changes to the branch - @alfred accept!

**Comment by zp-alfred** [comment]
Accepted by robbiegill!

**Pull request merged**